frontend/packages/otr/src/redux/selectors/clerkInterpreter.ts

```diff
@@ -8,145 +7,98 @@ import { PermissionToPublish } from 'enums/interpreter';
-            currentDate
-          )
-        )
-      )
-    );
-
-    return { effective, expiring, expired };
-  }
-);
-
 export const selectFilteredClerkInterpreters = createSelector(
   (state: RootState) => state.clerkInterpreter.interpreters,
   (state: RootState) => state.clerkInterpreter.filters,
   (interpreters, filters) => {
-    let filtered = interpreters;
+    let filteredInterpreters = interpreters;
 
     if (StringUtils.isNonBlankString(filters.name)) {
-      filtered = filtered.filter((interpreter) =>
-        filterByName(interpreter, filters.name as string)
+      filteredInterpreters = filteredInterpreters.filter((interpreter) =>
+        interpreterNameMatchesName(interpreter, filters.name as string)
       );
     }
 
-    filtered = filtered.filter((interpreter) =>
-      filterByQualificationCriteria(interpreter, filters)
+    return filteredInterpreters.filter((interpreter) =>
+      hasQualificationsMatchingFilters(interpreter, filters)
     );
-
-    return filtered;
   }
 );
 
-const filterByQualificationCriteria = (
+// Interpreter level predicates
+
+const interpreterNameMatchesName = (
+  { lastName, nickName }: ClerkInterpreter,
+  name: string
+) => {
+  const nameCombs = [`${nickName} ${lastName}`, `${lastName} ${nickName}`];
+
+  return nameCombs.some((comb) =>
+    comb.toLowerCase().includes(name.toLowerCase().trim())
+  );
+};
+
+const hasQualificationsMatchingFilters = (
   { qualifications }: ClerkInterpreter,
   filters: ClerkInterpreterFilters
 ) => {
-  const currentDate = dayjs();
-
-  return qualifications.find(
-    (q) =>
-      matchesLanguageFilters(filters, q) &&
-      matchesExaminationType(filters, q) &&
-      matchesPermissionToPublish(filters, q) &&
-      matchesQualificationStatus(filters, q, currentDate)
-  );
+  const matchesFilters = (qualifications: Array<Qualification>) => {
+    return qualifications.filter(
+      (q) =>
+        matchesFromLang(filters, q) &&
+        matchesToLang(filters, q) &&
+        matchesExaminationType(filters, q) &&
+        matchesPermissionToPublish(filters, q)
+    );
+  };
+
+  switch (filters.qualificationStatus) {
+    case QualificationStatus.Effective:
+      return matchesFilters(qualifications.effective).length > 0;
+    case QualificationStatus.Expiring:
+      return matchesFilters(qualifications.expiring).length > 0;
+    case QualificationStatus.Expired:
+      return matchesFilters(qualifications.expired).length > 0;
+    case QualificationStatus.ExpiredDeduplicated:
+      return matchesFilters(qualifications.expiredDeduplicated).length > 0;
+  }
 };
 
-// Qualification matchers
+// Qualification level predicates
+
+const matchesFromLang = (
+  { fromLang }: QualificationFilter,
+  qualification: Qualification
+) => (fromLang ? fromLang == qualification.fromLang : true);
+
+const matchesToLang = (
+  { toLang }: QualificationFilter,
+  qualification: Qualification
+) => (toLang ? toLang == qualification.toLang : true);
```

Comment:
Tämä taitaa nyt toimia eri tavalla kuin aiemmassa versiossa? Aiemmassa oli seuraava kommentti:
```
// Qualifications are bidirectional. A "from"/"to" filter
// should match either language of a qualification.
```

Reply:
Toimii eri lailla kuin aiemmin joo. Muistan ihan tarkoituksella muuttaneeni koodia tähän suuntaan, että `fromLang` ja `toLang` perusteilla haettaisiin erikseen, mutta nyt kun ajatuksella tätä mietin uudestaan, niin kyllä tä itse asiassa pitäisi mennä näin kun on ollut aiemminkin.

Ajatus mistä tän koodin muuttaminen kumpusi johtui siitä, että ajattelin tän koodin olevan vanhaa perua ajalta, jolloin frontendin piti varautua siihen, että "Mistä" valintakentän arvot ovat ihan mitä tahansa kieliä eikä vain kotimaisia kieliä (parhaillaan vain `suomi`). Tämä muuttui myöhemmin tosiaan niin että "Mistä" valinnat vaihtuivat kaikista tulkkien kieliparien kielistä vain kotimaisiksi (suomi) kieliksi. Mutta toki sitten kun joskus jollain löytyy esim. ruotsi-englanti rekisteröinti, niin loogista, että mahdollisuus hakea valinnalla ruotsi-suomi yhtä lailla kuin suomi-ruotsi (ellei tuota haluta jollain tavoin estää). Näin ruotsi-suomi pitäisi palauttaa saman tuloksen kuin suomi-ruotsi ja nää muutokset kannattaisi varmaan palauttaa tosiaan ennalleen.

Luulin että `publicInterpreter` alla ei tätä kaksisuuntaista hakua olisi enää tuettukaan, mutta näemmä sieltä löytyy myös. Palautetaan tämä vaan siis ennalleen.

Reply:
Korjattu